### ISSUE TITLE
Resources: Purpose/Quantity/Overload

### ISSUE THREAD

**Issue by LibSpit** [We-the-People-civ4col-mod/Mod#180]
So I have mentioned this a couple times, and I see a few others mention it as well, so I thought I would open a discussion about it. (Not sure how to add tags yet...)

Topics of Discussion: (feel free to expand to other topics I haven't thought of)
Are there too many resources?
Are there too many 'make to sell' resources? (are their prices too similar to be worth it.)
Do varied resources need varied 'abilities'? (like cloth and rope making ships, etc.)
Possible solutions/ideas/improvements to the current situation.

Are there too many resources?
I guess my answer would probably be yes, at least from the perspective of starting out, it is just an information overload when you first open the city screen of your first colony. Especially when you take into account topic number 2.

Are there too many 'make to sell' resources?
This was the one that got me, particularly when I realised that so many of them are virtually the same price. It quickly became apparent that most of them were really just the same thing.

Now I can accept the argument that different climates need different resources for balance. However it just got to the point where it was a lot 'mess' trying to see the relevant information, so many icons to learn and discern all at once.

Do varied resources need varied 'abilities'?
I think this is a yes, rather than all resources having the same ability (like being worth 5 gold) I feel like they need some varied point. Some reason to try and gain/supply/control each one. Now the ability can just be value. Like train oil(no train sad face) being worth a lot of money (same as silver!? Wow when I found that I bee lined to whale boats!).

I know some things ended in a combined resource (I am thinking the cloth and dye lines making coloured cloth.) which I like, X+Y samey resource = new super resource that is better than both.

However that seemed to be an exception rather than a rule.

I feel like this needs to be looked at, with the question "why do I NEED this resource" especially in the context of so many things seeming to do more or less the same thing within 1-3 gold of each other.  

Possible solutions/ideas/improvements to the current situation.
A lot of these thoughts are going back over things that Night and I (and probably some others) have talked about over the span of time...

1.
Hiding/Revealing Yields on the city screen over time. (can also be done on the map too like Civ 4 does with some resources, Iron/Oil/etc.) So in the beginning you maybe only see the raw resources, with the rest either locked or the yield bar expands over time. 
These could be revealed by a tech tree or some form of era progression (like DoaNE where you complete goals to advance to the next era) 

2. I think I had some other thoughts but have temporarily forgotten them :D

**Comment by Nightinggale** [comment]
Unlocking yields have been mentioned in #62. Unlocking yields can easily be worked into CivEffects #93. CivEffects can be granted though various ways, including eras, meaning progress an era and more yields shows up. In fact feel free to say how you want it to be able to trigger and CivEffects will likely be able to either do it or gain the ability to trigger like that.

It's a good topic to bring up. There are a bunch of "produce to sell" yields and usually it doesn't really matter which you produce. In fact you can end up benefiting from focusing exclusively on one as it makes better usage of a constant flow with transports. That's not really the goal for having multiple yields. #144 would counter the transport slot argument though.

**Comment by NoFreeUsernameDammit** [comment]
I think it could do with removing a couple of resources. IMO best would be to remove luxury goods and coca, and move coffee to savannah hills (it's actually endemic to tropical highlands). It wouldn't be bad to give some more resources strategic value, although right now I can't imagine how exactly would that go.

Another issue that I keep running into in my games is storage capacity in late game. At 2-plot radius, the quanity of goods produced by a fully developed colony is roughly 500-1000 per turn. If a colony serves as export hub for several inland colonies, it quickly becomes swamped as it cannot hold even a turn's production, and no matter how well do you handle things logistically, the custom house goes into overflow mode and sells even important goods that you want to keep like tools, guns, horses...

**Comment by LibSpit** [comment]
Yeah that was what I found part way through my play through. I was trying to make use of all these resources, then at one point I just sort of came to a conclusion that it was pointless/ineffective to do so, then I found myself just looking at a very confusing/muddling yield bar, constantly trying to find the yields that applied to my situation.

**Comment by LibSpit** [comment]
I guess the running out of max space is just a balance issue and is fairly easy to solve by jut adding some more warehouse tiers.

**Comment by orlanth** [comment]
These are good points about the perils of adding MOAR yields for their own sake (and have given me pause re adding too many yields to 2071 mod - see overcomplicated design doc [here](https://docs.google.com/spreadsheets/d/11CDOqSoocXZ0OI9ZIDF0dqEhknbOh4R8NCXSdMzSKME/edit#gid=0) :  ) 🤣 Here are a few game design strategies that might be helpful to make games with multiple yield types more interesting and less same-y:

1. **Allow Techs to gradually unlock** (and potentially obsolete) Yields as Lib mentioned: some cheap early yields eventually giving way to more rare/profitable ones that are more difficult to produce.

2. **Add unique in-game strategic uses** for each: in M:C, one neat feature is that researching Techs can consume one or more yield types based on the Tech being researched. Consuming specific goods for particular areas of research could add interesting strategic tradeoffs beyond mere price when considering what to seek out and produce - you might even want to import specific inputs in certain circumstances. Other ways to do this include adding more requirements for specific yields on building/unit productions or Profession adoptions. CON: Vanilla AI is already poor at planning ahead for existing yield requirements for buildings/units or Professions it may want to adopt in future. This may require a civeffect handicapinfos cheat reducing/exempting yield requirements to make AI competitive, or allowing it to use yields within Plotgroups to avoid requiring it to plan supply and transport for multiple possible yields and requirements too far in advance for it to cope effectively.

3. **Enhance supply/demand price effects**, increase price variances between different cities and tradescreens. If local prices decrease quickly in response to oversupply of a specific yield and increase when a specific yield is undersupplied, prices between different cities and tradescreens become much more dynamic and interesting. If you're making too much easy money overproducing yield X, gradually decreasing profitability will counterbalance this and make you seek out new markets or new Yields to exploit.

4. **Increased use for Domestic Markets** : if more Units and Buildings can generate demand for specific yields, it would be interesting to track what % of local demand in cities is satisfied per turn, and grant some reward for efficiently satisfying your citizens' demands. (M:C had a city-wide variable called Prosperity similar to Rebel Sentiment, something like this could be used to track your progress in this). CON: could lead to wagon train micromanagement hell if you need to physically schlep each yield type across your empire - to offset this, a Civeffect could allow domestic market yield use within Plotgroups to represent a more natural free-market system as seen in games like Victoria 2.

**Comment by ShadesOT** [comment]
>  I was trying to make use of all these resources, then at one point I just sort of came to a conclusion that it was pointless/ineffective to do so, ...

I try to satisfy the domestic market in my games and it usually makes me stinking rich (ending in batch buying ships of the line in europe ... ).

> ... the custom house goes into overflow mode and sells even important goods that you want to keep like tools, guns, horses ...

What about setting the minimum quanitity, until which the good is not sold? The customs house screen is there to solve this very problem.

> ... goods produced by a fully developed colony is roughly 500-1000 per turn. If a colony serves as export hub for several inland colonies, it quickly becomes swamped as it cannot hold even a turn's production, ...

That is actually true pretty often and can be improved with larger warehouse capacities.

Still I want to take the occasion, to tell about my solution, I am somewhat proud of. For me the key was NOT to use hub cities but to use transportation circles.
I had carriages only ship goods between 2 cities each. (This is far more efficient than anything the transportation AI ever comes up with.) And I built a designated customs house export city. All overflow gets dumped there and I don't care about it anymore. Looking at the domestic market goods, the transportation system connects all cities in one big circle, with the export city included. The production of one city is propagated to the following cities in the circle. The import/export settings in each city are set to import unlimited and keep only some amount to satisfy the domestic market. If there is so much production, that the threshhold in the export city is overflown, it starts selling and keeps the amount of goods in the circle at a steady level. The import demand in the city, following the export city is the only one, which has to be adjusted additionally periodically. And of course more production means slowly more carriages are needed. In this setup the goods are mostly inside the carriages, on turn end, not in the warehouses. An additional benefit: If the transport circle for domestic market goods runs clockwise and the transport circle for raw yields runs counter clockwise, one carriage can work both circles. Also, as a carriage only travels between two cities, new cities can easily be linked into the transportation circles. And it can be done with everything. Luxuries and stone get injected into the circle at an import city. Lumber and tools propagate from cities with overproduction to where they are needed.
The downsides are the number of carriages needed, as stuff is transported all the time in a "take all the overproduction from city 1 and ask in city 2,3,4 ... if it is needed"-style. And depending on the number of cities in the circle, the propagation of goods from where there are too many to where there are to little can be slow. The latter problem can be reduced by breaking bigger circles into smaller ones, with the export cities as connecting nodes. Additionally, nothing hinders me to set up special routes to meet special demands.

> In essence if there are 5 raw resources that are all worth the same could some of them be amalgamated into a more generic name like 'Trade Crops' or 'Luxury Crops'. Just a thought if the diversification is really cosmetic only and has no real game play variation...

I agree with you mostly. There is a logistic challenge, handling that many goods though. And the cosmetics are really good looking. But if we throw that away, to implement more interesting game mechanics, I am all for it.


> So as an example you might be able to build one or two factories in a city, that can produce one of a selection of items, but two or three items may be needed to make a higher value item in the future, so it would mean a new settlement with a new factory being supplied with the old produced goods.

This one I do not like so much. Scarcity of building slots? Come on! To players this always feels like superimposed restrictions with no justification. I tend to think, that such a mechanic is implemented into games, where the designers didn't find anything better, to create a challenge. IMO making factories very expensive and introduce maintainance cost and overhead which, in the worst case, can yield a negative overall outcome, would be the way to go in this regard.

IMO interesting gameplay does not come from hierarchical game mechanics - the more the better/the higher the better (point for your thoughts, regarding the multitude of raw and manufactured goods) and superimposed restrictions on the height or the width. Interesting gameplay comes from game mechnics, which counter each other. And in the best case scenario, which do not have an equilibrium that fits all scenarios best, but with multiple equilibria, each being best, depending on the situation in the nation. Like state of exploration, expansion, consolidation, small cities, big cities, war with natives, union with natives, war with europeans, war of independence, land empires, sea empires, trading empires, small empires, big ones, etc ... .
An additional problem in civ4 in general is, that everything built or commissioned is of the "fire and forget" type. A building stands, has not maintenance cost, doesn't decay, when unused, just adds a positive modifier. At least 1/2 of the founding fathers provide positive effecets forever after (which also do not counter each other). A pioneer, once commissioned, can spam raods and improvements forever (at least they cost money). And so on.

My summarized opinion on the topic is  ... WTP as continuation of RAR and TAC has all those yields, which are fun to look at and are a challenge to manage logistically. I am all for giving up 3/4 ths of them, if we come up with some other challenging game mechanic, or - even better - find game mechanics, which make the production of all those yields an interesting necessity (like here https://github.com/We-the-People-civ4col-mod/Mod/issues/166#issuecomment-429648713). But until that day, keep them and make the warehouses bigger if desired. The re-implementation of the railroad (let's please do that in a non hierarchical and non fire-and-forget way) and maybe the big river mechanic will also aid logistics. Not to mention, that we could implement tools, which provide more oversight regarding the logistics and wich, in the best case, enable to manage all the logistics from one point/screen.

**Comment by LibSpit** [comment]
Yeah my biggest 'concern' with the current amount of yields was that it gave me a headache (But this is just a personal issue) but I persevered thinking, ok lets get to grip with what all these things do. Until I realised that many of them didn't really do anything. I then found myself slowly losing motivation as I had this big, 'messy' interface of yields that I was mostly trying to ignore rather than make use of, because they were all basically the same value/thing.  

The logistical challenge of managing all these resources would be fun, if I felt some point to engaging in the challenge, but I struggled to find any reason to bother. One simple solution to this might be making more interesting diversity of prices in the three trade screens.

I noticed that horses did well in the non-eurozone so I started to try and build a purely African trade route so send my horses down. If this was the same for other resources, Like Booze and rope and sailcloth do well in Port Royale, spices do well in Africa and the luxurious items do well in Europe, I would start to see a real logistics challenge and value in the resources as now I need to have ships that sail to each zone carrying specific cargo. (Maybe this is already a thing, but I looked and didn't really get a sense of that)

Manufacturing/Scarcity of Building Slots.
My point was not so much a scarcity of building slots without justification, (I could make arguments as to the 'sense' of it) it was more along the idea of a shift in the 'Yield Purpose' so instead of having a tonne of raw yields, it was instead shifting over to produced yields and supply chains for increasing the value of the finished product. Only having one factory (or factory/industrial district) is a way to make spreading out and creating supply chains more of a 'thing' rather than having a mega city that just builds everything and no real logistics is required (beyond maybe food and raw material). 
(The fact is the mega city becomes a 'city centre' colony and then outlying 'suburban colonies').

This is coming from M:C where we had things like 'pop cap.' for cities and in order to increase it you had to build housing and plumbing and things of that nature.

I could go into a list of things that make limited 'building slots' for manufacturing a reasonable thing and the many ways that it could be expanded etc. with as you say 'counter balancing' concepts and all that jazz, That is not really the point of this topic though. I first brought up the factory/production concept simply as another 'purpose' for yields, or as a shifting in weight of yield types. 
So more of a complexity in the production side rather than in a number of chains that produce negligible differences in value throughout the process.

**Comment by LibSpit** [comment]
Added a new thought (No. 4) to the Original Post.

**Comment by abmpicoli** [comment]
Some ideas.
1) Most of the goods (including some vanilla), shouldn't be available both to natives and simple europeans : they are sugar (asian origin), coffee (african origin), grapes (european origin). They will only be available for production if there is at least one expert planter at the colony: the rationale is that to adapt the crops to the new world was not a trivial task. Specially coffee... https://en.wikipedia.org/wiki/History_of_coffee . https://en.wikipedia.org/wiki/History_of_sugar . https://en.wikipedia.org/wiki/History_of_wine .

Natives won't be able to produce those goods as well..

2) Make the market have meaningful attrition... The goods are all the same because the market for them is always the same, with small variation... Make europe have stocks and consumption of goods.... Make that random events (such as droughts damaing europe crops) would make some goods increase in price (until the stock is re-filled)... So identify a proper commodity for sale becomes a meaningful part of the game.

Example, Europe have 5000 food in storage... a random event claiming that crops were damage happens, lowering it's stock to 0. The food price goes up, making food a commodity worth of being sold to europe... Make the same for Africa and Port Royal.

The price should be 1+MAX(0,<<maximum storage>> - EuropeanStock  / <<maximum storage>>) * Maximum_Price ;

MaximumStorage=<<european consumption>>*30 .

Every ~30 turns (we could make it random, say, 1 chance out of 30 of consumption pattern to change ) , european consumption patterns will change, representing changes in consuming patterns (new fads, new sources of that product). A random value between -MaxChange and +MaxChange 

The market should have a special "job" there: merchants would generate "market trend" points. If there is enough "market trend" points, the europe consuption of goods would be visible.

**Comment by Nightinggale** [comment]
`
They will only be available for production if there is at least one expert planter at the colony
`
An implementation of this would be like the current school system. Have a BoolArray in CvCity and yields can only be produced if the array index is true or unit is expert. Once an expert has been there for a turn, the city has learned how to handle the yield in question.

During new city creation, set the array to the values from a default array. This array is calculated post xml loading and contains true for any yield where no unit can teach the colony how to produce the yield in question.

As for gameplay, I'm less certain I like it. I fear it can become more annoying than fun. CivEffects can provide the same effect, but on a per player level, meaning less micromanagement. According to the CivEffect roadmap, events will be able to grant a CivEffect, either permanently or multi turn events. I can imagine something like "if expert on map, grant event, which enables yield". It's much simpler to manage for the player and it gives sort of the same result: yields are unavailable until the expert arrives.

**Comment by abmpicoli** [comment]
> As for gameplay, I'm less certain I like it. I fear it can become more annoying than fun. CivEffects can provide the same effect, but on a per player level, meaning less micromanagement. According to the CivEffect roadmap, events will be able to grant a CivEffect, either permanently or multi turn events. I can imagine something like "if expert on map, grant event, which enables yield". It's much simpler to manage for the player and it gives sort of the same result: yields are unavailable until the expert arrives. 
I'm ok with that...

**Comment by LibSpit** [comment]
DOANE has the Agronomist system, where you find seeds in the new world and you have an agronomist plant and develop them, as well as crops have optimal latitudes for growth. That is pretty fun system for raw yields.

**Comment by vissercsp** [comment]
...Well, would it be an idea that when starting the game the player can choose which resources can be grown and cultivated? This solves the problem for those who want less resources and for those that love to have a ton of them.

Personally, I like a lot of them - the more the better. Especially when these are required to fabricate something else, such as hemp for sails and rope. And considering the fact that the "new world" had a lot to offer the current palette available is very much to my likings. Actually, there are to that I miss: coal (a major source of fuel, especially for industries) and tea (otherwise no Boston Tea Party... ;-)  ).

**Comment by raystuttgart** [comment]
**Comment on "choosing Ressources / Yields to be grown and cultivated":**
Very bad idea technically. It will be much to difficult considering all the dependencies (Professions, Experts, Buildings, Bonus Ressources at Map Generation, Bonus Ressources on Scenario Maps, ...)
It will simply not be manageable in Balancing either.
I also do not feel that people complain because we have too many Yields / Ressources.
That is what most WTP players like.

**Comment on "making Yields more interesting / varied":**
We are working on concepts to do that.
(e.g. Happiness, improving Health, improving Domestic Market, ...)

Just a small comment on **tea**:

Tea was not culitvated in any considerable amount in the New World.
(And even today it is not.)

**Tea** was imported from England (which itself imported it from Asia).
The Boston Tea Party was in no way related to destroying goods the New World had produced itself - they destroyed goods imported from the Old World. (Tea was very expensive due to very high taxes that directly went into the pockest of the King and East India Trading Company.)

Just a small comment on **coal**:

**Coal** was not really important in the early 15th and 16th century even though yes it was used for producing Iron and later Steal. It only became so profitable in the age of industrialization which however started a couple of centuries later.

**Comment by ErikSandbraaten** [comment]
Coal would fit in if we expanded the timeline to include the industrial age. Of course, a much wanted imperialism mod-mod could also do this :)

**Comment by raystuttgart** [comment]
Well, but we cannot included Yields / Ressources in WTP core mod (for colonial age time line) for a mod-mod that uses a different time line.
That needs to be done by the mod-mod itself.

If we however find good reasons or arguments (in terms of a good feature concept that fits the colonial age time line) to include coal into WTP I would be ok with it.

I simply do not want create a mod-mod in the core mod itself.
